compiler/astalgo.nim

```diff
@@ -407,15 +414,29 @@ proc key(this: var DebugPrinter; key: string) =
   this.res.add "\": "
 
 proc value(this: var DebugPrinter; value: string) =
+  if this.useColor:
```

Comment:
You can put an argument to add a macro to `terminal.nim` that does the wrapping of color/style and reset code around strings.